src/main/java/com/coveros/selenified/element/check/State.java

```diff
@@ -0,0 +1,314 @@
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + CHECKED, timeTook, Success.FAIL);
+        }
+        return !isChecked;
+    }
+
+    /**
+     * Verifies that the element is editable. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. If the element isn't an input, this will
+     * constitute a failure, same as it not being editable. This information
+     * will be logged and recorded, with a screenshot for traceability and added
+     * debugging support.
+     */
+    void editable();
+
+    default boolean checkEditable(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + EDITABLE, waitFor);
+        // verify the element is present
+        if (!isPresent(waitFor)) {
+            return false;
+        }
+        // perform the check
+        boolean isEditable = getElement().is().editable();
+        // record the result
+        if (!isEditable) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTEDITABLE, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + EDITABLE, timeTook, Success.PASS);
+        }
+        return isEditable;
+    }
+
+    /**
+     * Verifies that the element is not editable. If the element isn't present,
+     * it waits up to the default time (5 seconds) for the element, before
+     * marking this verification as a failure. If the element isn't an input,
+     * this will constitute a pass, as non input elements are not editable. This
+     * information will be logged and recorded, with a screenshot for
+     * traceability and added debugging support.
+     */
+    void notEditable();
+
+    default boolean checkNotEditable(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + NOTEDITABLE, waitFor);
+        // verify the element is present
+        if (!isPresent(waitFor)) {
+            return false;
+        }
+        // perform the check
+        boolean isEditable = getElement().is().editable();
+        // record the result
+        if (!isEditable) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTEDITABLE, timeTook, Success.PASS);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + EDITABLE, timeTook, Success.FAIL);
+        }
+        return !isEditable;
+    }
+
+    /**
+     * Verifies that the element is enabled. If the element isn't present, it waits
+     * up to the default time (5 seconds) for the element, before marking this
+     * verification as a failure. This information will be logged and recorded, with
+     * a screenshot for traceability and added debugging support.
+     */
+    void enabled();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "enabled" to prevent any misunderstanding/clash with field "ENABLED" defined on line 48. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)